curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -189,32 +207,43 @@ int64_t DiskCacheManager::CacheDiskUsedRatio() {
     int64_t usedPercent = 100 * usedBytes / (usedBytes + availableBytes) + 1;
 
     VLOG(3) << "cache disk usage = " << usedPercent;
+
+    diskFsUsedRatio_.store(usedPercent, std::memory_order_seq_cst);
     return usedPercent;
 }
 
-bool DiskCacheManager::IsDiskCacheFull() {
-    int64_t ratio = CacheDiskUsedRatio();
-    if (ratio < 0) {
-        LOG(ERROR) << "get disk use ratio error.";
-        return false;
+void DiskCacheManager::SetDiskInitUsedBytes() {
+    std::string cmd = "timeout " + std::to_string(cmdTimeoutSec_) +
+        " du -sb " + cacheDir_ + " | awk '{printf $1}' ";
+    SysUtils sysUtils;
+    std::string result = sysUtils.RunSysCmd(cmd);
+    if (result.empty()) {
+        LOG(ERROR) << "get disk used size failed.";
+        return;
     }
-    if (ratio >= fullRatio_) {
+    usedBytes_.fetch_add(std::stod(result), std::memory_order_seq_cst);
```

Comment:
std::stod will throw an exception when “result” is invalid， use another function in curve/common instead